File: app/src/main/java/br/edu/farol/gadoplus/storage/database/preference/PreferenceManager.java

package br.edu.farol.gadoplus.storage.database.preference;


import android.content.Context;
import android.content.SharedPreferences;


public class PreferenceManager {

    private static final String FILE_NAME = "preferences";
    private static final String FIRST_START = "first_start";
    private static SharedPreferences mPreferences;
    public static void initialize(Context context) {
        mPreferences = context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);
    }

    public static void setIsFirstStartDone(boolean done) {
        mPreferences.edit().putBoolean(FIRST_START, done).apply();
    }

    public static boolean isFirstStartDone() {
        return mPreferences.getBoolean(FIRST_START, false);
    }
}

File: app/src/main/java/br/edu/farol/gadoplus/storage/database/preference/PreferenceManager.kt
package br.edu.farol.gadoplus.storage.database.preference

import android.content.Context
import android.content.SharedPreferences

object PreferenceManager {
    private const val FILE_NAME = "preferences"
    private const val FIRST_START = "first_start"
    private var mPreferences: SharedPreferences? = null
    fun initialize(context: Context) {
        mPreferences = context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)
    }

    var isFirstStartDone: Boolean
        get() = mPreferences!!.getBoolean(FIRST_START, false)
        set(done) {
            mPreferences!!.edit().putBoolean(FIRST_START, done).apply()
        }
}